sdk/ai/Azure.AI.Projects/samples/Sample17_FineTuning_DPO.md

```diff
@@ -0,0 +1,143 @@
+string validationFilePath = Environment.GetEnvironmentVariable("VALIDATION_FILE_PATH") ?? "data/dpo_validation_set.jsonl";
+var endpoint = Environment.GetEnvironmentVariable("PROJECT_ENDPOINT");
+var modelDeploymentName = Environment.GetEnvironmentVariable("MODEL_DEPLOYMENT_NAME");
+AIProjectClient projectClient = new AIProjectClient(new Uri(endpoint), new DefaultAzureCredential());
+ProjectOpenAIClient oaiClient = projectClient.OpenAI;
+OpenAIFileClient fileClient = oaiClient.GetOpenAIFileClient();
+FineTuningClient fineTuningClient = oaiClient.GetFineTuningClient();
+```
+
+## Upload Training and Validation Files
+
+To fine-tune a model, we need to upload training and validation datasets. In the code below, we use the `UploadFile` method of `OpenAIFileClient`. This method returns an `OpenAIFile` object containing the file ID and `Status`, which indicates whether the file was successfully uploaded to the cloud. We use these to monitor the upload process in the `WaitForFileProcessing` and `WaitForFileProcessingAsync` helper methods.
+
+### Async
+
+```C# Snippet:AI_Projects_FineTuning_DPO_UploadFilesAsync
+// Upload training file
+Console.WriteLine("Uploading training file...");
+using FileStream trainStream = File.OpenRead(trainingFilePath);
+OpenAIFile trainFile = await fileClient.UploadFileAsync(
+    trainStream,
+    "dpo_training_set.jsonl",
+    FileUploadPurpose.FineTune);
+Console.WriteLine($"Uploaded training file with ID: {trainFile.Id}");
+
+// Upload validation file
+Console.WriteLine("Uploading validation file...");
+using FileStream validationStream = File.OpenRead(validationFilePath);
+OpenAIFile validationFile = await fileClient.UploadFileAsync(
+    validationStream,
+    "dpo_validation_set.jsonl",
+    FileUploadPurpose.FineTune);
+Console.WriteLine($"Uploaded validation file with ID: {validationFile.Id}");
+```
+
+### Sync
+
+```C# Snippet:AI_Projects_FineTuning_DPO_UploadFiles
+// Upload training file
+Console.WriteLine("Uploading training file...");
+using FileStream trainStream = File.OpenRead(trainingFilePath);
+OpenAIFile trainFile = fileClient.UploadFile(
+    trainStream,
+    "dpo_training_set.jsonl",
+    FileUploadPurpose.FineTune);
+Console.WriteLine($"Uploaded training file with ID: {trainFile.Id}");
+
+// Upload validation file
+Console.WriteLine("Uploading validation file...");
+using FileStream validationStream = File.OpenRead(validationFilePath);
+OpenAIFile validationFile = fileClient.UploadFile(
+    validationStream,
+    "dpo_validation_set.jsonl",
+    FileUploadPurpose.FineTune);
+Console.WriteLine($"Uploaded validation file with ID: {validationFile.Id}");
+```
+
+## Wait for File Processing
+
+In production, you should wait for files to complete processing before creating a fine-tuning job. See the helper methods in `Sample16_FineTuning_Supervised.md` for `WaitForFileProcessingAsync` and `WaitForFileProcessing` implementations.
```

Comment:
Let us add the `WaitForFileProcessing` for all sample to make sure that the sample is always passing.